rest/ProductService.js

```diff
@@ -82,11 +88,21 @@ class ProductService {
             throw new IllegalArgumentError(INVALID_QUANTITY_ERR);
         }
     }
+    
 
-    /**
-     * TODO
-     * Validate serial numbers
-     */
+    #assertProductIsActive(productId) {
```

Comment:
Shouldn't this be reversed?

asseertProductIsActive means we want it to be active. Rename assertProductIsInactive / assertProductIsNotActive